### PR TITLE
Integrated zod with backend endpoints

### PR DESCRIPTION
**Pull request by JamesColesanti**

### CODE REVIEW

**Comment by rymaju** [comment]
@JamesColesanti Looks great but there are some package scoping things to figure out
- `node_modules` and the common `package.json`s should be in the `/common` folder
- that local package needs to be linked by the other two packages

**Comment by rymaju** [comment]
Since this can be a bit finicky let me figure out how to do this with `npm link`... unless you want to give it a shot first

This will likely break _something_ in CD, just don't know what

File: common/api.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+const userApi = makeApi(
+  [
+    {
+        method: "get",
+        path: "/",
+        alias: "", 
+        description: "Home Page",
+        response: z.object({ ok: z.date() }),
+    },
```
Comment (rymaju) [inline common/api.ts:12]: 💭 Not sure if we should extract or inline the endpoint zod objects themselves

File: common/api.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            phoneNumber: z.string().length(11),
+            website: z.string(),
+            languages: z.string(),
+            modality: z.string(),
+            businessLocation: z.string(),
+            businessName: z.string(),
+            minAgeServed: z.number(),
+            email: z.string().email(),
+            fullName: z.string()
+        })),
```
Comment (rymaju) [inline common/api.ts:29]: This definitely should be extracted into a `/dto` or `/schema` folder and exported individually

File: frontend/src/SearchTherapists.tsx [hunk 3]
```diff
@@ -153,7 +153,7 @@ export const SearchTherapists: React.FC = () => {
                               fontSize='xs'
                               textTransform='uppercase'
                             >
-                              Webiste
+                              Website
```
Comment (rymaju) [inline frontend/src/SearchTherapists.tsx:156]: 😅 nice catch

**Comment by JamesColesanti** [comment]
> @JamesColesanti Looks great but there are some package scoping things to figure out
> 
> * `node_modules` and the common `package.json`s should be in the `/common` folder
> * that local package needs to be linked by the other two packages

Assuming this has been done already, but just curious, could you explain a little more what this means? Specifically, what the "local" package and the "other two packages" are exactly?

**Pull request closed without merging**